File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/project_settings/project/ProjectSettings.java
package com.github.alexandrecarlton.idea.settings.layout.project_settings.project;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableProjectSettings.class)
public interface ProjectSettings {

  Optional<String> projectName();

  Optional<String> projectSdk();

  Optional<ProjectLanguageLevel> projectLanguageLevel();

}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/project_settings/project/ProjectSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.project_settings.project

import com.fasterxml.jackson.annotation.JsonProperty

data class ProjectSettings(

    @JsonProperty("Project name")
    val projectName: String? = null,

    @JsonProperty("Project SDK")
    val projectSdk: String? = null,

    @JsonProperty("Project language level")
    val projectLanguageLevel: ProjectLanguageLevel? = null
)
